app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -0,0 +1,92 @@
+module ForemanHostReports
+  module ReportsDashboardHelper
+
+    def searchable_links(name, search, counter)
+      # TODO: implement global filtering
+      content_tag :li do
+        content_tag(:span, raw('&nbsp;'), :class => 'label', :style => "background-color:" + report_color[counter]) +
+          raw('&nbsp;') +
+          link_to(name, hosts_path(:search => search), :class => "dashboard-links") +
+          content_tag(:h4, totals[counter])
+      end
+    end
+
+    def report_color
+      {
+        :change_hosts => "#4572A7",
+        :failure_hosts => "#AA4643",
+        :nochange_hosts => "#DB843D",
+        :disabled_hosts => "#92A8CD",
+      }
+    end
+
+    def get_overview(report, options = {})
+      state_labels = {
+        change_hosts: _('Change'),
+        failure_hosts: _('Failure'),
+        nochange_hosts: _('No Change'),
+        disabled_hosts: _('Disabled alerts'),
+      }
+      state_filters = {
+        change_hosts: report_changes.count,
+        nochange_hosts: report_nochanges.count,
+        failure_hosts: report_failures.count,
+        disabled_hosts: disabled_alerts.count
+      }
+
+      {
+        data: state_labels.map { |key, label| [label, report[key], report_color[key]] },
+        searchUrl: hosts_path(search: '~VAL~'),
+        searchFilters: state_filters.each_with_object({}) do |(key, filter), filters|
+          filters[state_labels[key]] = filter
+        end,
+      }
+    end
+
+    def totals
+      @report = {} if @report.nil?
+      @report.update(
+        :total_hosts => Host.all.count,
+        :change_hosts => report_changes.count,
+        :nochange_hosts => report_nochanges.count,
+        :failure_hosts => report_failures.count,
+        :disabled_hosts => disabled_alerts.count
+      )
+    end
+
+    def latest_reports
+      @latest_reports ||= HostReport.authorized(:view_host_reports).my_reports
+                                    .where(host_id: hosts.pluck(:id))
+                                    .reorder(:last_report => :desc)
+                                    .limit(9)
+                                    .preload(:host)
+    end
+
+    def latest_reports?
+      latest_reports.limit(1).present?
+    end
+
+    private
+
+    def disabled_alerts
+      hosts.where(:enabled => false)
+    end
+
+    def report_changes
+      Host.joins(:host_reports).where("change > 0 ").uniq
+    end
+
+    def report_nochanges
+      Host.joins(:host_reports).where("nochange > 0").uniq
+    end
+
+    def report_failures
+      Host.joins(:host_reports).where("failure > 0").uniq
+    end
```

Comment:
Can you make all the four methods actually to return count? That is safer, we never want to load all the records into memory so let's not even return it.